Isodose/Logic/vtkSlicerIsodoseModuleLogic.cxx

```diff
@@ -62,14 +62,19 @@
 #include <vtkTransformPolyDataFilter.h>
 #include <vtkTriangleFilter.h>
 #include <vtkWindowedSincPolyDataFilter.h>
+#include <vtkAppendPolyData.h>
+#include <vtkPointData.h>
+#include <vtkUnsignedCharArray.h>
+#include <vtkFloatArray.h>
+
 #include "vtksys/SystemTools.hxx"
 
 //----------------------------------------------------------------------------
 const char* DEFAULT_ISODOSE_COLOR_TABLE_FILE_NAME = "Isodose_ColorTable.ctbl";
 const char* DEFAULT_ISODOSE_COLOR_TABLE_NODE_NAME = "Isodose_ColorTable_Default";
 const char* DEFAULT_ISODOSE_COLOR_TABLECOPY_NODE_NAME = "Isodose_ColorTable_DefaultCopy";
 const char* RELATIVE_ISODOSE_COLOR_TABLE_NODE_NAME = "Isodose_ColorTable_Relative";
-const std::string vtkSlicerIsodoseModuleLogic::ISODOSE_MODEL_NODE_NAME_PREFIX = "IsodoseLevel_";
+const std::string vtkSlicerIsodoseModuleLogic::ISODOSE_MODEL_NODE_NAME_PREFIX = "IsodoseLevels";

```

Comment:
This is not used as a prefix any more although the name suggests so. I think it would be good to use it as a name postfix, i.e. you append "_IsodoseLevels" to the dose volume name so that they can be differentiated when there is more than one dose (the current way all of them have the exact same name). Make sure you overwrite any existing one.

Reply:
> Make sure you overwrite any existing one.
So it must be only one isodose model node per dose volume?

Reply:
That was how it has worked so far. It is very easy to generate an isodose set, and it is not useful to have them parallel I think, so yes.
@lassoan @Sunderlandkyl what do you think?

Reply:
When there were many output isodose models then things were already complicated so it made some sense to not make things even more complicated by having multiple sets of isodose models per dose volume.

Now that we will have a single output model, we can follow the usual output node selection pattern: have a node selector for the output model and overwrite that node. If the user wants to create/update another output then he can create/choose a different output model. This would make the Isodose module behave like most other modules.

Reply:
Yes, I think it looks good. Maybe we could make the "Generated isodose" button a but larger (wider, higher). This button looks like a QToolButton (which is much more tight around the text) and changing it to a QPushButton could make it a bit larger, more prominent.

The output isodose model could have a "None" option with custom None text "Create new isodose model", so if the user leaves it at default then a new model is created (after the model is created it would be set as output model).

Tiny remark: we generally try to avoid using "node" on the user interface, could you replace "Isodose model node" to "Model node"?

Reply:
It would be nice to add the unit to the legend. I would also make things look a bit more pleasing if the size of 5, 10, 15, ... numbers would be a bit smaller and the title of the colorbar would be displayed closer to the bar.

Reply:
> I would also make things look a bit more pleasing if the size of 5, 10, 15, ... numbers would be a bit smaller and the title of the colorbar would be displayed closer to the bar.

That is a bug in color legend actor when label == text.

Reply:
Yes, indeed there seems to be some unnecessary cross-talk between width of a vertical color legend and the height of the colorbar: if the width is increased then the height of the colorbar is becomes wider and shorter. Overall, the behavior seems to be about the same, regardless of what is used as label (value or color name). Not a big issue, not urgent to fix it, but at some point it would be nice to investigate this.

Reply:
I've added color legend defaults, when color legend is created for isodose model for the fist time.

![image](https://user-images.githubusercontent.com/3785912/161916294-04289ff1-06d6-47f0-a07c-639a68fec71e.png)

Reply:
This looks nice. If you load images from DICOM and slice view annotations are displayed then the top-left and bottom-left corners can contain many lines of patient information. There may be some space on the left side near the middle, but if we want to avoid overlapping of patient information and color legend then it would be safer to display color legend on the right side by default.

Reply:
Position was changed to the right side of the screen.

Reply:
Color legend in the right side of the screen by default.

![image](https://user-images.githubusercontent.com/3785912/163352748-dc38065b-3724-4d29-9a98-a2309437fcdb.png)